c7n/resources/ebs.py

```diff
@@ -79,12 +80,77 @@ class resource_type(object):
     action_registry = ActionRegistry('ebs-snapshot.actions')
 
     def resources(self, query=None):
+        qfilters = QueryParser.parse(self.data.get('query', []))
         query = query or {}
+        if qfilters:
+            query['Filters'] = qfilters
         if query.get('OwnerIds') is None:
             query['OwnerIds'] = ['self']
         return super(Snapshot, self).resources(query=query)
 
 
+class QueryParser(object):
+
+    QuerySchema = {
+        'description': six.string_types,
+        'owner-alias': ('amazon', 'amazon-marketplace', 'microsoft'),
+        'owner-id': six.string_types,
```

Comment:
QuerySchema should also allow for `OwnerIds` as an attribute per lines 87-88

```python
       if query.get('OwnerIds') is None:
            query['OwnerIds'] = ['self']
```

Reply:
Not sure that it should, the vast majority are self owned, querying public snaps isn’t useful, since their isn’t anything to do with them from a custodian perspective,  it’s setup this way to add due to use with non owned amis.

Reply:
in that case should we remove owner-id from the set of filters then? otherwise seems like it could result in confusion

Reply:
though I do agree on not really needing to query public snapshots,  was trying to rack my brain if there were valid usecases

Reply:
so in this context, the query parser schema is specific to what can be passed in Filters not generically to all parameters on describe. owner-ids can be specified for the policy query and it will be passed through.

Reply:
gotcha, 👍